easybuild/easyconfigs/m/matplotlib/matplotlib-3.5.1-foss-2020b.eb

```diff
@@ -0,0 +1,65 @@
+easyblock = 'PythonBundle'
+
+name = 'matplotlib'
+version = '3.5.1'
+
+homepage = 'https://matplotlib.org'
+description = """matplotlib is a python 2D plotting library which produces publication quality figures in a variety of
+ hardcopy formats and interactive environments across platforms. matplotlib can be used in python scripts, the python
+ and ipython shell, web application servers, and six graphical user interface toolkits."""
+
+toolchain = {'name': 'foss', 'version': '2020b'}
+
+builddependencies = [
+    ('pkg-config', '0.29.2'),
+]
+
+dependencies = [
+    ('Python', '3.8.6'),
+    ('SciPy-bundle', '2020.11'),
+    ('libpng', '1.6.37'),
+    ('freetype', '2.10.3'),
+    ('Tkinter', '%(pyver)s'),
+    ('Pillow', '8.0.1'),
+]
+
+use_pip = True
+sanity_pip_check = True
+
+# avoid that matplotlib downloads and builds its own copies of freetype and qhull
```

Comment:
I see that this is new compared to our existing EasyConfigs for matplotlib. Did matplotlib introduce some change in the build mechanism that now changes this? I see for one of my prior versions that it was correctly linked with my system qhull/freetype:
```
/sw/arch/Debian10/EB_production/2021/software/matplotlib/3.4.2-foss-2021a/lib64/python3.9/site-packages/matplotlib$ ldd _qhull.cpython-39-x86_64-linux-gnu.so
        linux-vdso.so.1 (0x00007ffeef7c5000)
        /sara/tools/xalt/xalt/lib64/libxalt_init.so (0x0000152b6d3c1000)
        libqhull_r.so.8.0 => /sw/arch/Debian10/EB_production/2021/software/Qhull/2020.2-GCCcore-10.3.0/lib/../lib64/libqhull_r.so.8.0 (0x0000152b6d347000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x0000152b6d15d000)
        libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x0000152b6d156000)
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x0000152b6cfd3000)
        /lib64/ld-linux-x86-64.so.2 (0x0000152b6d413000)
sw/arch/Debian10/EB_production/2021/software/matplotlib/3.4.2-foss-2021a/lib64/python3.9/site-packages/matplotlib$ ldd ft2font.cpython-39-x86_64-linux-gnu.so
        linux-vdso.so.1 (0x00007ffe32690000)
        /sara/tools/xalt/xalt/lib64/libxalt_init.so (0x000014e9225cd000)
        libfreetype.so.6 => /sw/arch/Debian10/EB_production/2021/software/freetype/2.10.4-GCCcore-10.3.0/lib/libfreetype.so.6 (0x000014e922514000)
        libstdc++.so.6 => /sw/arch/Debian10/EB_production/2021/software/GCCcore/10.3.0/lib64/libstdc++.so.6 (0x000014e92232d000)
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x000014e92217f000)
        libgcc_s.so.1 => /sw/arch/Debian10/EB_production/2021/software/GCCcore/10.3.0/lib64/libgcc_s.so.1 (0x000014e922165000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x000014e921fa4000)
        libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x000014e921f9f000)
        libbz2.so.1.0 => /sw/arch/Debian10/EB_production/2021/software/bzip2/1.0.8-GCCcore-10.3.0/lib64/libbz2.so.1.0 (0x000014e921f89000)
        libpng16.so.16 => /sw/arch/Debian10/EB_production/2021/software/libpng/1.6.37-GCCcore-10.3.0/lib64/libpng16.so.16 (0x000014e921f4c000)
        libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x000014e921f2b000)
        libz.so.1 => /sw/arch/Debian10/EB_production/2021/software/zlib/1.2.11-GCCcore-10.3.0/lib64/libz.so.1 (0x000014e921f12000)
        /lib64/ld-linux-x86-64.so.2 (0x000014e92262e000)
        ```

Reply:
Not sure what you mean being new?  I took the latest [matplotlib-3.4.3-foss-2021b.eb](https://github.com/easybuilders/easybuild-easyconfigs/blob/develop/easybuild/easyconfigs/m/matplotlib/matplotlib-3.4.3-foss-2021b.eb),  compared with Spack builds and upgraded missing pip lib. Version 3.5.1 also brings  PySide6 compatibility and new template config naming. That's all regarding the build.

Reply:
Hm, it popped up when I did `eb --review-pr 15095`, which does a multidiff against older versions. Let me check again...

Reply:
Ok, my bad: the `--review-pr` for some reason compared to the `2020b` versions. Those didn't have this section yet. But indeed, `2021a` (on which my ldd was based) does. So it makes sense that it was resolved there, because that EasyConfig would have included the `_fix_setup` :) Ok, we keep this, clearly :)